rs/nns/governance/src/voting.rs

```diff
@@ -94,25 +237,80 @@ struct ProposalVotingStateMachine {
     recent_neuron_ballots_to_record: BTreeMap<NeuronId, Vote>,
 }
 
-impl ProposalVotingStateMachine {
-    fn try_new(proposal_id: ProposalId, topic: Topic) -> Result<Self, String> {
-        if topic == Topic::Unspecified {
-            return Err("Topic must be specified".to_string());
+impl From<ProposalVotingStateMachine> for crate::pb::v1::ProposalVotingStateMachine {
+    fn from(value: ProposalVotingStateMachine) -> Self {
+        Self {
+            proposal_id: Some(value.proposal_id),
+            topic: value.topic as i32,
+            neurons_to_check_followers: value.neurons_to_check_followers.into_iter().collect(),
+            followers_to_check: value.followers_to_check.into_iter().collect(),
+            recent_neuron_ballots_to_record: value
+                .recent_neuron_ballots_to_record
+                .into_iter()
+                .map(|(n, v)| (n.id, v as i32))
+                .collect(),
         }
+    }
+}
 
+impl TryFrom<crate::pb::v1::ProposalVotingStateMachine> for ProposalVotingStateMachine {
+    type Error = String;
+
+    fn try_from(value: crate::pb::v1::ProposalVotingStateMachine) -> Result<Self, Self::Error> {
         Ok(Self {
+            proposal_id: value.proposal_id.ok_or("Proposal ID must be specified")?,
+            topic: Topic::try_from(value.topic).map_err(|e| e.to_string())?,
+            neurons_to_check_followers: value.neurons_to_check_followers.into_iter().collect(),
+            followers_to_check: value.followers_to_check.into_iter().collect(),
+            recent_neuron_ballots_to_record: value
+                .recent_neuron_ballots_to_record
+                .into_iter()
+                .map(|(n, v)| {
+                    let neuron_id = NeuronId::from_u64(n);
+                    let vote = Vote::try_from(v).map_err(|e| e.to_string())?; // Propagate the error directly
+                    Ok((neuron_id, vote))
+                })
+                .collect::<Result<_, Self::Error>>()?,
+        })
+    }
+}
+
+impl Storable for crate::pb::v1::ProposalVotingStateMachine {
+    fn to_bytes(&self) -> Cow<'_, [u8]> {
+        Cow::from(self.encode_to_vec())
+    }
+
+    fn from_bytes(bytes: Cow<'_, [u8]>) -> Self {
+        Self::decode(&bytes[..])
+            // Convert from Result to Self. (Unfortunately, it seems that
+            // panic is unavoid able in the case of Err.)
+            .expect("Unable to deserialize ProposalVotingStateMachine.")
+    }
+
+    const BOUND: Bound = Bound::Unbounded;
+}
+
+impl ProposalVotingStateMachine {
+    fn new(proposal_id: ProposalId, topic: Topic) -> Self {
+        Self {
             proposal_id,
             topic,
             ..Default::default()
```

Comment:
Nit: I think initializing each field is often better. In the future a field might be added where default is not the right choice

Reply:
Agreed, done.